frontend/catalyst/programs/verification.py

```diff
@@ -185,3 +192,57 @@ def _obs_checker(obs):
         _obs_checker(obs)
 
     return (tape,), lambda x: x[0]
+
+
+@transform
+def validate_observables(
+    tape: QuantumTape, qjit_capabilities: dict, name: str
+) -> (Sequence[QuantumTape], Callable):
+    """Validates the observables and measurements for a circuit against the capabilites
+    from the TOML file.
+
+    Args:
+        tape (QuantumTape or QNode or Callable): a quantum circuit.
+        qjit_capabilities (dict): specifies the capabilities of the qjitted device
+        name (str): the name of the device to use in error messages.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[.QuantumTape], function]:
+        The unaltered input circuit.
+
+    Raises:
+        CompileError: if an observable is not supported by the device with Catalyst
+
+    """
+
+    def _observable_is_supported(obs) -> bool:
+        """Specifies whether or not an observable is accepted by QJITDevice.
+
+        If the observable is built on one or multiple other observables, check
+        both that the overall observable is supported, and that its component
+        parts are supported."""
+
+        if isinstance(obs, Tensor):
+            return all([_observable_is_supported(o) for o in obs.obs])
+
+        if isinstance(obs, (Hamiltonian, CompositeOp, SymbolicOp)):
+
+            if not qjit_capabilities.native_obs.get(obs.name):
+                return False
+
+            if hasattr(obs, "operands"):
+                return all([_observable_is_supported(o) for o in obs.operands])
+            elif hasattr(obs, "ops"):
+                return all([_observable_is_supported(o) for o in obs.ops])
+            elif hasattr(obs, "base"):
+                return _observable_is_supported(obs.base)
```

Comment:
Nice, might be useful to add a tiny inline comment which property is from which type of observable

Reply:
done